crates/nu-protocol/src/config/mod.rs

```diff
@@ -48,7 +49,7 @@ impl Default for HistoryConfig {
 
 #[derive(Serialize, Deserialize, Clone, Debug)]
 pub struct Config {
-    pub external_completer: Option<usize>,
+    pub external_completer: Option<Closure>,
```

Comment:
This makes more sense for type safety but slightly grows `Config` that is prone to cloning.

Reply:
True, should I add an `Arc` here?

Reply:
If the number of captures is low, I am not sure if this is really the straw that breaks the `Config`s back. I think we can figure out separately on which level of granularity we gain the most from arcing